typescript/web/src/components/pagination-toolbar/pagination-toolbar.tsx

```diff
@@ -0,0 +1,100 @@
+import {
+  useColorModeValue as mode,
+  Text,
+  Grid,
+  HStack,
+  Center,
+} from "@chakra-ui/react";
+import { ToggleButtonGroup, ToggleButton } from "../toggle-button-group";
+import { PaginationTool } from "../pagination-tool";
+
+interface PaginationToolbarProps {
+  itemsTotalCount: number;
+  itemsPerPageOptions: number[];
+  currentPage: number;
+  itemsPerPage: number;
+  onChange: (page: number, itemsPerPage: number) => void;
+}
+
+export const PaginationToolbar = (props: PaginationToolbarProps) => {
+  const {
+    itemsTotalCount,
+    itemsPerPageOptions,
+    currentPage,
+    itemsPerPage,
+    onChange: setPagesAndItemsPerPages,
+  } = props;
+
+  const itemsPerPageStrToNum = itemsPerPageOptions.reduce<
+    Record<string, number>
+  >((record, num) => {
+    return {
+      ...record,
+      [num.toString()]: num,
+    };
+  }, {} as Record<string, number>);
+
+  const skip = (currentPage - 1) * itemsPerPage;
+  const maxPages = Math.ceil(itemsTotalCount / itemsPerPage);
+
+  if (maxPages && currentPage > maxPages) {
+    setPagesAndItemsPerPages(maxPages, itemsPerPage);
+  }
+
+  return (
+    <Grid
+      pos="fixed"
+      bottom="0px"
+      w="100%"
+      h="44px"
```

Comment:
Unless necessary, try to use Chakra values instead of directly pixels (same thing goes for everywhere you're using pixel values)